sdk/cosmos/azure_data_cosmos/src/models/container_properties.rs

```diff
@@ -0,0 +1,468 @@
+use std::time::Duration;
+
+use azure_core::Model;
+use serde::{Deserialize, Deserializer};
+
+use crate::models::SystemProperties;
+
+#[cfg(doc)]
+use crate::clients::ContainerClientMethods;
+
+fn deserialize_ttl<'de, D>(deserializer: D) -> Result<Option<Duration>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    Ok(Option::<u64>::deserialize(deserializer)?.map(Duration::from_secs))
+}
+
+/// Properties of a Cosmos DB container.
+///
+/// Returned by [`ContainerClient::read()`](crate::clients::ContainerClient::read()).
+#[non_exhaustive]
+#[derive(Model, Clone, Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct ContainerProperties {
+    /// The ID of the container.
+    pub id: String,
+
+    /// The time-to-live for items in the container.
+    ///
+    /// For more information see <https://docs.microsoft.com/azure/cosmos-db/time-to-live#time-to-live-configurations>
+    #[serde(default)]
+    #[serde(deserialize_with = "deserialize_ttl")]
+    pub default_ttl: Option<Duration>,
+
+    /// The time-to-live for the analytical store in the container.
+    ///
+    /// For more information see <https://docs.microsoft.com/azure/cosmos-db/analytical-store-introduction#analytical-ttl>
+    #[serde(default)]
+    #[serde(deserialize_with = "deserialize_ttl")]
+    pub analytical_storage_ttl: Option<Duration>,
+
+    /// The definition of the partition key for the container.
+    pub partition_key: PartitionKeyDefinition,
+
+    /// The indexing policy for the container.
+    pub indexing_policy: Option<IndexingPolicy>,
+
+    /// The unique key policy for the container.
+    pub unique_key_policy: Option<UniqueKeyPolicy>,
+
+    /// The conflict resolution policy for the container.
+    pub conflict_resolution_policy: Option<ConflictResolutionPolicy>,
+
+    /// The vector embedding policy for the container.
+    pub vector_embedding_policy: Option<VectorEmbeddingPolicy>,
+
+    /// A [`SystemProperties`] object containing common system properties for the container.
+    #[serde(flatten)]
+    pub system_properties: SystemProperties,
+}
+
+/// Represents the vector embedding policy for a container.
+#[non_exhaustive]
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct VectorEmbeddingPolicy {
+    /// The [`VectorEmbedding`]s that describe the vector embeddings of items in the container.
+    #[serde(rename = "vectorEmbeddings")]
+    pub embeddings: Vec<VectorEmbedding>,
+}
+
+/// Represents the vector embedding policy for a container.
+#[non_exhaustive]
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct VectorEmbedding {
+    /// The path to the property containing the vector.
+    pub path: String,
+
+    /// The data type of the elements stored in the vector.
+    pub data_type: VectorDataType,
+
+    /// The number of dimensions in the vector.
+    pub dimensions: u64,
```

Comment:
We're in the process of changing this to `u32` elsewhere.  It was a C++ long -> C#/Java long that caused initial confusion.